frontend/src/Dashboard/HistoricData.js

```diff
@@ -92,79 +110,100 @@ const formatData = (data) => {
       icon: VscDashboard,
       titleText: 'Szélsebesség',
       text: data.windData.windSpeed,
-      unit: 'km/h"',
+      unit: 'km/h',
     },
-
+  ];
+};
+const soilData = (data) => {
+  return [
     {
       icon: WiThermometer,
       titleText: 'Talaj hőmérséklet 0cm',
-      text: data.soilData.soilTemperature0cm,
+      text: data.soilData.soilTemperature0cm.toFixed(2),
       unit: <>&#8451;</>,
     },
 
     {
       icon: GiDrop,
       titleText: 'Talaj nedvesség 30cm',
-      text: data.soilData.soilMoisture30cm,
+      text: data.soilData.soilMoisture30cm.toFixed(2),
       unit: 'V/V %',
     },
     {
       icon: GiDrop,
       titleText: 'Talaj nedvesség 60cm',
-      text: data.soilData.soilMoisture60cm,
+      text: data.soilData.soilMoisture60cm.toFixed(2),
       unit: 'V/V %',
     },
 
     {
       icon: GiDrop,
       titleText: 'Talaj nedvesség 90cm',
-      text: data.soilData.soilMoisture90cm,
+      text: data.soilData.soilMoisture90cm.toFixed(2),
       unit: 'V/V %',
     },
 
     {
       icon: GiDrop,
       titleText: 'Talaj nedvesség 120cm',
-      text: data.soilData.soilMoisture120cm,
+      text: data.soilData.soilMoisture120cm.toFixed(2),
       unit: 'V/V %',
     },
-    {
-      icon: HiSun,
-      titleText: 'Besugárzás',
-      text: data.miscData.irradiation,
-      unit: 'W/m2',
-    },
-
-    {
-      icon: HiSun,
-      titleText: 'Fényegység',
-      text: data.miscData.lightUnit,
-      unit: 'cd',
-    },
-
+  ];
+};
+const batteryData = (data) => {
+  return [
     {
       icon: RiBattery2ChargeLine,
       titleText: 'Napelem töltő feszültség',
-      text: data.batteryData.solarCellChargingVoltage,
+      text: data.batteryData.solarCellChargingVoltage.toFixed(2),
       unit: 'V',
     },
 
     {
       icon: GiCarBattery,
       titleText: 'Külső akkufeszültség',
-      text: data.batteryData.externalBatteryVoltage,
+      text: data.batteryData.externalBatteryVoltage.toFixed(2),
       unit: 'V',
     },
 
     {
       icon: GiCarBattery,
       titleText: 'Belső akkufeszültség',
-      text: data.batteryData.internalBatteryVoltage,
+      text: data.batteryData.internalBatteryVoltage.toFixed(2),
       unit: 'V',
     },
   ];
 };
 
+const windDirection = (wind) => {
+  if (wind < 22.5 || wind > 337.5) {
+    return 'Északi';
+  } else if (wind < 67.5) {
+    return 'Észak-Keleti';
+  } else if (wind < 112.5) {
+    return 'Keleti';
+  } else if (wind < 157.5) {
+    return 'Dél-Keleti';
+  } else if (wind < 202.5) {
+    return 'Déli';
+  } else if (wind < 247.5) {
+    return 'Dél-Nyugati';
+  } else if (wind < 292.5) {
+    return 'Nyugati';
+  } else if (wind < 337.5) {
+    return 'Észak-Nyugati';
```

Comment:
(Lásd lentebb.) Egészítsétek ki minimum egy else ággal, ami kezeli a hibát, de a jó megoldás az lenne ha az elején checkolnátok undefined-ra, invalid értékekre.

Reply:
Javitva, az elején levizsgálva.

Reply:
Nem jo meg mindig, most a null es az undefined, vagy >360 || <0 is Eszak-Nyugati szel lesz, holott nyilvanvaloan az adattal nem stimmel valami. A hibat kezelni kell, informalni a usert a hianyarol, nem pedig ad hoc behelyettesiteni egy default value-val.

Reply:
Javitotttam, remelem igy mostmar jo!